pkg/controllers/thanos-querier/components.go

```diff
@@ -0,0 +1,170 @@
+package thanos_querier
+
+import (
+	"fmt"
+
+	monv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	msoapi "github.com/rhobs/monitoring-stack-operator/pkg/apis/v1alpha1"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type components []client.Object
+
+func thanosComponents(thanos *msoapi.ThanosQuerier, sidecarUrls []string) components {
+	name := "thanos-querier-" + thanos.Name
+	resourceLabels := map[string]string{
+		"app.kubernetes.io/part-of": name,
+	}
+
+	return []client.Object{
+		// ServiceAccount
+		ensureLabels(newServiceAccount(name, thanos.Namespace), resourceLabels),
+		// Deployment
+		ensureLabels(newThanosQuerierDeployment(name, thanos, sidecarUrls), resourceLabels),
+		// Service
+		ensureLabels(newService(name, thanos.Namespace), resourceLabels),
+		ensureLabels(newServiceMonitor(name, thanos.Namespace), resourceLabels),
+	}
+}
+
+func newThanosQuerierDeployment(name string, spec *msoapi.ThanosQuerier, sidecarUrls []string) *appsv1.Deployment {
+	args := []string{
+		"query",
+		"--grpc-address=127.0.0.1:10901",
+		"--http-address=127.0.0.1:9090",
+		"--log.format=logfmt",
+		"--query.auto-downsampling",
+	}
+	for _, endpoint := range sidecarUrls {
+		// args = append(args, fmt.Sprintf("--endpoint=%s", endpoint))
```

Comment:
I left it as a reminder since the next thanos release will deprecate the `--store` argument in favor of the `--endpoint` argument. See https://github.com/thanos-io/thanos/blob/78d250bbb0856fefa9e21dd340ef0b82cabb87f5/cmd/thanos/query.go#L111